lib/layers/data/repositories/achieves_repo_impl.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:wot_statistic/generated/l10n.dart';
+import 'package:wot_statistic/layers/data/models/local/user_data.dart';
+import 'package:wot_statistic/layers/data/models/remote/achievements_data/achievement_data.dart';
+import 'package:wot_statistic/layers/data/models/remote/achievements_data/achievements_database.dart';
+import 'package:wot_statistic/layers/data/models/remote/user_achieves/user_achieves_api_data.dart';
+import 'package:wot_statistic/layers/data/sources/local/achieves_local_datasource.dart';
+import 'package:wot_statistic/layers/data/sources/remote/remote_data_source.dart';
+import 'package:wot_statistic/layers/domain/entities/achieves.dart';
+import 'package:wot_statistic/layers/domain/repositories/achieves_repo.dart';
+
+const epic = "epic";
+const action = "action";
+const special = "special";
+const memorial = "memorial";
+const group = "group";
+const achieveClass = "class";
+
+class AchievesRepoImpl implements AchievesRepo {
+  final AchievesLocalDataSource achievesLocalDataSource;
+  final RemoteDataSource remoteSource;
+
+  const AchievesRepoImpl({
+    required this.achievesLocalDataSource,
+    required this.remoteSource,
+  });
+
+  UserData get signedUser {
+    final _signedUser = achievesLocalDataSource.getSignedUser();
+    if (_signedUser == null) throw Exception(S.current.SignedUserIsNotExist);
+    return _signedUser;
+  }
+
+  @override
+  Future<List<List<Achieve>>> fetchAchieves() async {
+    await _initAchievesDatabase();
+    final UserAchievesApi achievesApi;
+    try {
+      achievesApi =
+          await remoteSource.fetchAchievesData(accountId: signedUser.id);
+    } catch (e) {
+      throw Exception(S.current.CheckInternetConnection);
+    }
+    final Map<String, int> achievesId =
```

Comment:
To improve readability of you code and have ability to cover it with test, please move such peaces of code to separate method. It will be much easier understand what you are doing there, if of this part will be moved to fun and have name